pipeline/execout/cachev1/engine.go

```diff
@@ -0,0 +1,143 @@
+package cachev1
+
+import (
+	"context"
+	"fmt"
+	"github.com/streamingfast/bstream"
+	"github.com/streamingfast/dstore"
+	pbsubstreams "github.com/streamingfast/substreams/pb/sf/substreams/v1"
+	"github.com/streamingfast/substreams/pipeline/execout"
+	"go.uber.org/zap"
+	"regexp"
+)
+
+var cacheFilenameRegex *regexp.Regexp
+
+func init() {
+	cacheFilenameRegex = regexp.MustCompile(`([\d]+)-([\d]+)\.output`)
+}
+
+type Engine struct {
+	ctx               context.Context
+	caches            map[string]*OutputCacheState
+	SaveBlockInterval uint64
+	baseCacheStore    dstore.Store
+	logger            *zap.Logger
+}
+
+type OutputCacheState struct {
+	c           *OutputCache
+	initialized bool
+}
+
+func NewEngine(ctx context.Context, saveBlockInterval uint64, baseCacheStore dstore.Store, logger *zap.Logger) (execout.CacheEngine, error) {
+	e := &Engine{
+		ctx:               ctx,
+		caches:            make(map[string]*OutputCacheState),
+		SaveBlockInterval: saveBlockInterval,
+		baseCacheStore:    baseCacheStore,
+		logger:            logger,
+	}
+	return e, nil
+}
+
+func (e *Engine) Prime(modules map[string]*pbsubstreams.Module) error {
```

Comment:
This function knows too much about the `Module`, where it only needs a mapping between Name and Hash. Don't we have that with the ModuleHashes object?

That's the only concern of the CacheEngine.

I'd decouple from `pbsubstreams.Module`

Reply:
Wondering about `Prime`, what we prime? It's not prime number related either.

Isn't that simply an `Init`ialization?